### PR TITLE
Reduce the burden to run Azure Pipelines and make Windows CI run faster

### PR DESCRIPTION
**Pull request by jcwchen**
### Description

To reduce the burden for running Azure Pipelines and make Windows CI run faster: 
- Don't run AzurePipelines again in the main branch. In the merge queue (`gh-readonly-queue/**`) it has already run once additionally before merging into the main branch.
- Increases the max parallel jobs in settings from 13 to 20.
- Decreases the total jobs for each Azure Pipelines from 5 to 4. So ideally onnx repo should be able to run 5 PRs at the same time. (20/4=5)
- Make Windows CI run faster: do not test Protobuf 3.20 when the test config is internal

Other updates:
- Increases the timeout for merge queue from 90 mins to 300 mins (thanks @prasanthpul!)
- Removes cancel-in-progress: true in release CIs (which was brought by #5431), because the pipelines will just be skipped if there is a newer commit coming: https://github.com/onnx/onnx/commits/main. I think it will make future debug harder. Revert the changes in this PR as well.

This PR helps to resolve the timeout issue in the merge queue.

### Motivation and Context

https://github.com/onnx/onnx/pull/5431 helped some PRs got merged, but there are still some pipeline removed from merge queue due to timeout. Let's try to improve the traffic in AzurePipelines.

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -14,10 +14,6 @@ on:
 permissions:  # set top-level default permissions as security best practice
   contents: read
 
-concurrency:
-  group: ${{ github.workflow }}-${{ github.ref }}-${{ github.event_name == 'workflow_dispatch' }}
-  cancel-in-progress: true
-
```
Comment (justinchuby) [inline .github/workflows/lint.yml:20]: We can keep this so PRs updates do not trigger an excessive amount of lint jobs

File: .azure-pipelines/Linux-CI.yml [hunk 2]
```diff
@@ -1,5 +1,4 @@
 trigger:
-- main
```
Comment (justinchuby) [inline .azure-pipelines/Linux-CI.yml:2]: Do we need at least _some_ guards on the main branch? Are the release CIs doing this?

**Review by justinchuby — APPROVED** [review]

File: .github/workflows/lint.yml (hunk 1 above)
Comment (jcwchen) [reply .github/workflows/lint.yml]: Good point. Lint CI is a fast one and it shouldn't be skipped frequently in the main branch.

File: .azure-pipelines/Linux-CI.yml (hunk 2 above)
Comment (jcwchen) [reply .azure-pipelines/Linux-CI.yml:2]: Yes, release CIs will still be running in the main branch.

File: .azure-pipelines/Windows-CI.yml [hunk 3]
```diff
@@ -62,6 +56,7 @@ jobs:
       git submodule update --init --recursive
       set ONNX_BUILD_TESTS=1
       set ONNX_ML=$(onnx_ml)
+      set ONNX_VERIFY_PROTO_3=$(onnx_verify_proto)
```
Comment (jcwchen) [inline .azure-pipelines/Windows-CI.yml:59]: It might just be accidentally removed somehow... Adding it back to cover more cases.

**Pull request merged**